.github/workflows/coverage_runner.yml

```diff
@@ -0,0 +1,109 @@
+    if: github.event_name == 'push' || needs.check_for_membership.outputs.check-result == 'true' || github.event_name == 'workflow_dispatch'
+    name: Run tests with Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.10' ]
+        os: [ ubuntu-latest, windows-latest ]
+      fail-fast: false
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+          
+      - name: Checkout code for push event
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+
+      - name: Checkout code for PR
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          
+      - name: Checkout repository to fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v2
+        
+      - name: Fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git fetch origin pull/${{github.event.inputs.pr_number }}/head:temp_branch
+          git checkout temp_branch
+          
+      - name: Install dependencies
+        run: |
+          pip install -r requirements-test.txt
+      - name: Run tests
+        env:
+          HAZELCAST_ENTERPRISE_KEY: ${{ secrets.HAZELCAST_ENTERPRISE_KEY }}
+        run: python run_tests.py
+            
+      - name: Publish results to Codecov for PR coming from hazelcast organization
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' &&  github.event_name == 'pull_request_target' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          override_pr: ${{ github.event.pull_request.number }}
+          
+      - name: Publish results to Codecov for Push
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' &&  github.event_name == 'push' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          
+      - name: Publish result to Codecov for PR coming from community
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' && github.event_name == 'workflow_dispatch' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          override_pr: ${{ github.event.inputs.pr_number }}
+          
+      - name: Upload remote controller logs if test run fails
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: rc-logs-${{ matrix.python-version }}-${{ matrix.os }}
+          path: |
+            rc_stderr.log
+            rc_stdout.log
```

Comment:
Why did you deleted this part?

```
  run-linter:
    runs-on: ubuntu-latest
    name: Run black to check the code style
    steps:
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: '3.10'
      - name: Checkout to code
        uses: actions/checkout@v2
      - name: Install dependencies
        run: |
          pip install -r requirements-dev.txt
      - name: Run black
        run: |
          black --check --config black.toml .
  generate_docs:
    runs-on: ubuntu-latest
    name: Generate documentation
    steps:
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: '3.10'
      - name: Checkout to code
        uses: actions/checkout@v2
      - name: Install dependencies
        run: |
          pip install -r requirements-dev.txt
      - name: Generate documentation
        working-directory: docs
        run: |
          make html SPHINXOPTS="-W --keep-going -b linkcheck"
  run-mypy:
    runs-on: ubuntu-latest
    name: Run mypy to check type annotations
    steps:
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          python-version: '3.10'
      - name: Checkout to code
        uses: actions/checkout@v2
      - name: Install dependencies
        run: |
          pip install -r requirements-dev.txt
      - name: Run mypy
        run: |
          mypy --show-error-codes hazelcast
```

Reply:
As the new event trigger type is a little bit dangerous for security, I thought it will be good to use it when it is a need. For these jobs actually we don't need any secrets. I was planning to create new workflows for jobs which does not need any secrets. But you are right because if I am removing the old workflow, I should add the workflows I planned for these jobs.
Adding a new workflow for these jobs